### PR TITLE
Automate pypi release

### PR DESCRIPTION
**Pull request by urutva**
### Description

Automate pypi release

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

File: pypi-release.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  # Compare latest tag and a tag previous to that tag
+  - bash: |
+      latest_tag=`git tag --sort=-creatordate | sed -n '1p'`
+      previous_tag=`git tag --sort=-creatordate | sed -n '2p'`
+      actual_diff=`git diff $latest_tag $previous_tag`
+      if [ -z "$actual_diff" ]; then
+        echo There is no difference between last two commits namely $latest_tag and $previous_tag. Aborting release!!! >&2
+      fi
+    failOnStderr: true
+    displayName: "Check for changes in this release"
```
Comment (Patater) [inline pypi-release.yml:18]: What motivates checking for differences between release tags? I would have assumed a human made the new release tag to trigger a release to pypi. Why stop such a human when the release they tagged is identical to the previous one? Does it hurt anything to do a release with a new version number that's identical to a previous release?

File: pypi-release.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    displayName: 'Twine Authenticate ${{ parameters.repoName }}'
+    inputs:
+      pythonUploadServiceConnection: testpypi-${{ parameters.repoName }}
+
+  # Build the python distribution from source
+  - script: "python setup.py sdist bdist_wheel --universal"
+    displayName: "Prepare ${{ parameters.repoName }} for release"
+    workingDirectory: ${{ parameters.wDirectory }}
+
+  # Use command line script to 'twine upload', use -r to pass the repository name and --config-file to pass the environment variable set by the authenticate task.
```
Comment (Patater) [inline pypi-release.yml:30]:
```diff
- , use
+ . Use
```

File: src/mbed_os_tools/VERSION.txt [hunk 3]
```diff
@@ -1 +0,0 @@
-0.0.15
```
Comment (Patater) [inline src/mbed_os_tools/VERSION.txt:1]: Yay!

**Review by Patater — CHANGES_REQUESTED** [review]
Looks pretty good

File: pypi-release.yml (hunk 1 above)
Comment (urutva) [reply pypi-release.yml:18]: @Patater To ensure that same package is not released multiple times. The trigger behind the release automation is still the tag pushed by a human, but this step ensures *accidental* tags are not released to pypi. Why would one wants to release same package multiple times?

**Comment by urutva** [comment]
@jainvikas8 
> * There is a duplication of code when looking at [9250f39](https://github.com/ARMmbed/mbed-os-tools/commit/9250f3969be8f611ee93b1e6d14199eb7e318c46) , Can `setup(name='mbed-greentea',` be refactored from various setup.py as simple function with arguments?

The sub-packages `mbed-ls`, `mbed-host-tests` and `mbed-greentea` are released as independent packages to pypi. As far as I understand, we cannot have common function between these sub-packages.

> * Please remove `signed of` from commit messages

[Signed-off-by](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) typically certifies that committer has the rights to submit this work under the same license and agrees to a Developer Certificate of Origin. Can you please specify why it should be removed?

> * Why {WIP}?
Updated title.

File: pypi-release.yml (hunk 1 above)
Comment (Patater) [reply pypi-release.yml:18]: Seems odd to try to block this. If nothing breaks, and sombody makes and pushes a duplicate content tag on accident, one would expect a duplicate content release on pypi as well. Less code to maintain if we don't try to block stupidity and focus purely on mechanism and not policy.

**Comment by jainvikas8** [comment]
> > * Please remove `signed of` from commit messages
> 
> [Signed-off-by](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) typically certifies that committer has the rights to submit this work under the same license and agrees to a Developer Certificate of Origin. Can you please specify why it should be removed?

This is mostly to have consistency with other commits in this repo. I think if we are raising a PR here, then it will be decided to b accepted or not with the review process.

File: pypi-release.yml (hunk 1 above)
Comment (urutva) [reply pypi-release.yml:18]: @Patater OK. I'll remove `Check for changes in this release`

**Comment by urutva** [comment]
@Patater @jainvikas8 All the comments have been addressed. PR is ready for merge.

**Review by jainvikas8 — APPROVED** [review]

**Review by Patater — APPROVED** [review]

**Review by Patater — APPROVED** [review]

**Pull request merged**